src/main/java/graphql/servlet/context/DefaultGraphQLContextBuilder.java

```diff
@@ -1,24 +1,27 @@
-package graphql.servlet;
+package graphql.servlet.context;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.websocket.Session;
 import javax.websocket.server.HandshakeRequest;
 
+/**
+ * Returns an empty context.
+ */
 public class DefaultGraphQLContextBuilder implements GraphQLContextBuilder {
 
     @Override
     public GraphQLContext build(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
-        return new GraphQLContext(httpServletRequest, httpServletResponse);
+        return new DefaultGraphQLServletContext();
```

Comment:
Think it would be good if we would provide another interface layer in between, e.g. `GraphQLServletContext` and `GraphQLWebSocketContext` along with concrete implementations containing the respective objects. That way the `DefaultGraphQLServletContext` could still provide the `getFileParts()` for example, simplifying file uploads. And this way people could still make use of existing functionality if they simply grab a different type of context from the environment.

Do you agree? Happy to help out with this of course.

Reply:
I was curious as to how all that was intended to be used, and maybe I should've posted a question about it first. I also didn't run into any examples of it being used till trying to locally build the Spring boot project last night. I'm still digging though the commons project and trying to work out what I think, but I definitely don’t want to remove any functionality. Would it make more sense to wrap the result from the serlvet and allow it to be accessed there? I also ideally would like a way to break the dependency of the context builder on the actual http objects but I'm at a loss. My goal for keeping the graphql objects leaner is to make refactoring the servlet into a service layer easier. Maybe we can extend the execution input to maintain the file parts?

Reply:
Are you on Gitter by any chance? Might be easier to chat realtime :)

Couple of use cases I have used myself using the `HttpServletRequest` for the servlet and the `Session` for the WebSockets from the context:
* Authentication: usually getting the `Authorization` header to parse and populate the `SecurityContext` with the result. We could provide an alternate means for hooking in authentication that doesn't involve exposing the raw request through the context though.
* File uploads like already mentioned, so that could perhaps be expose differently too. You have to be able to get to them through the `DataFetchingEnvironment` in your method though. But perhaps a cleaner solution can be constructed for this which provides a nicer API for consumers.
* WebSocket Session also used for authentication.
* I use the `HttpServletRequest` as well to grab the remote IP address to verify subsequent calls come from the same IP address as where the JWT was generated for. Which is an additional security mechanism.

And these are just the ones I'm using, I obviously don't know what other use cases people might have atm...

Reply:
In the spring boot commons example, it looks like the mutation is intended to do something with the file parts (upload)? Should they be passed through the environment somehow instead?

Reply:
Don't believe you can, since the `DataFetchingEnvironment` is part of `graphql-java` itself and is agnostic and wants to remain agnostic of underlying technology. So they're not going to support file parts. The only thing under our control to pass along to the resolvers is in fact the `GraphQLContext` which can be retrieved from the `DataFetchingEnvironment`. That's why these they were exposed in the context, so the consumer could then in that resolver grab the files and handle them appropriately.